### PR TITLE
Add explicit logistic loss

### PR DESCRIPTION
**Pull request by EthanRosenthal**
This PR adds a logistic loss function for explicit, binary data. 

Due to interactions data typically being stored in sparse matrices prior to creation of the `Interactions` objects, we can't allow the target classes to be `(0, 1)`, as 0 would not be accessible as a value. I instead assume that the target classes will be `(-1, +1)`. The only weird part of this is that it then requires switching the observed ratings values to `(0, 1)` inside the loss function. While this not a big deal, the predictions end up offset by 1 compared to the ratings when running things like `model.predict()`.

### CODE REVIEW

File: spotlight/losses.py [hunk 1]
```diff
@@ -212,3 +212,33 @@ def poisson_loss(observed_ratings, predicted_ratings):
...
+    -------
+
+    loss, float
+        The mean value of the loss function.
+    """
+
+    assert_no_grad(observed_ratings)
+
+    # Convert target classes from (-1, 1) to (0, 1)
+    observed_ratings = (observed_ratings + 1) / 2
```
Comment (maciejkula) [inline spotlight/losses.py]: Could do `F.relu` here, might be faster.

File: spotlight/factorization/explicit.py [hunk 2]
```diff
@@ -142,6 +143,8 @@ def _initialize(self, interactions):
             self._loss_func = regression_loss
         elif self._loss == 'poisson':
             self._loss_func = poisson_loss
+        elif self._loss == 'logistic':
```
Comment (maciejkula) [inline spotlight/factorization/explicit.py]: Could you add this possibility to the class docstring above?

**Comment by EthanRosenthal** [comment]
I liked your idea of using `F.relu`, so I tried speed testing a couple different versions [here](https://gist.github.com/EthanRosenthal/ae1a6da04639292db28f345bec023b8a). The differences were negligible and I couldn't get the times to be reproducible. I ended up going with `torch.clamp` because it seemed the clearest from a readability standpoint.
